src/rosdiscover/observer/observer.py

```diff
@@ -0,0 +1,45 @@
+# -*- coding: utf-8 -*-
+__all__ = ("Observer",)
+import contextlib
+from abc import ABC, abstractmethod
+from typing import Dict, Iterator
+import roswire
+from roswire import AppInstance, ROSVersion
+
+from ..config import Config
```

Comment:
Since we don't actually need to interact with `Config` directly, we can do the old `if typing.TYPE_CHECKING` trick here. We will probably want to tweak `Config` to be a better OO class by adding `observer(...)` and `interpreter(...)` factory methods to it at some point in a future PR.

Reply:
Doing the above will also allow us to completely sidestep the nasty `ROS1Observer/ROS2Observer` import problem.